classic/original_autogpt/autogpt/utils/agent_discovery.py

```
@@ -0,0 +1,111 @@
+            return DiscoveryResult(cached_result) if cached_result else None
+
+    url = f"https://{domain}/.well-known/agent-discovery.json"
+    try:
+        req = urllib.request.Request(url, headers={"User-Agent": "agent-discovery/0.1"})
+        with urllib.request.urlopen(req, timeout=timeout) as resp:
+            if resp.status == 200:
+                data = json.loads(resp.read())
+                if use_cache:
+                    _cache[domain] = (time.time(), data)
+                return DiscoveryResult(data)
+    except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)